src/components/dialogs/reactive-limits/reactive-capability-curve/reactive-capability-utils.ts

```diff
@@ -129,10 +164,59 @@ export function setCurrentReactiveCapabilityCurveTable(
 }
 
-export function setCurrentReactiveCapabilityCurveChoice(
-    previousReactiveCapabilityCurveTable: ReactiveCapabilityCurve,
-    fieldKey: string,
-    setValue: UseFormSetValue<FieldValues>
+// In case the user has changed the choice of the reactive capability curve, but not the values, we keep the original choice
+export function toReactiveCapabilityCurveChoiceForGeneratorCreation(
+    currentReactiveLimits: GeneratorCreationDialogSchemaForm[typeof REACTIVE_LIMITS],
+    editData: GeneratorCreationInfos | null | undefined
 ) {
-    setValue(`${REACTIVE_LIMITS}.${REACTIVE_CAPABILITY_CURVE_CHOICE}`, 'MINMAX');
+    const currentChoice = currentReactiveLimits?.[REACTIVE_CAPABILITY_CURVE_CHOICE];
+    const previousChoice = editData ? (editData.reactiveCapabilityCurve === true ? 'CURVE' : 'MINMAX') : undefined;
+    return handleReactiveCapabilityCurveChoice(currentChoice, previousChoice, currentReactiveLimits);
+}
+
+// In case the user has changed the choice of the reactive capability curve, but not the values, we keep the original choice
+export function toReactiveCapabilityCurveChoiceForGeneratorModification(
+    currentReactiveLimits: GeneratorModificationDialogSchemaForm[typeof REACTIVE_LIMITS],
+    editData: GeneratorModificationInfos | null | undefined,
+    networkPoints: ReactiveCapabilityCurvePoints[] | null | undefined
+) {
+    const currentChoice = currentReactiveLimits?.[REACTIVE_CAPABILITY_CURVE_CHOICE];
+    const editDataChoice = editData
+        ? editData.reactiveCapabilityCurve?.value === true
+            ? 'CURVE'
+            : 'MINMAX'
+        : undefined;
+    const networkChoice = networkPoints ? 'CURVE' : 'MINMAX';
+    const previousChoice = editDataChoice ?? networkChoice;
+
+    return handleReactiveCapabilityCurveChoice(currentChoice, previousChoice, currentReactiveLimits);
+}
+
+function handleReactiveCapabilityCurveChoice(
+    currentChoice: string | null | undefined,
+    previousChoice: string | null | undefined,
+    currentReactiveLimits: GeneratorDialogSchemaBaseForm[typeof REACTIVE_LIMITS] | null | undefined
+) {
+    if (currentChoice === 'MINMAX') {
+        const hasAnyValue =
+            currentReactiveLimits?.[MAXIMUM_REACTIVE_POWER] || currentReactiveLimits?.[MINIMUM_REACTIVE_POWER];
+        if (previousChoice === 'CURVE' && !hasAnyValue) {
+            return 'CURVE';
+        } else {
+            return 'MINMAX';
+        }
+    } else if (currentChoice === 'CURVE') {
+        const currentPoints = currentReactiveLimits?.[REACTIVE_CAPABILITY_CURVE_TABLE];
+        const hasAnyValue = currentPoints?.some((v) => v.p != null || v.maxQ != null || v.minQ != null);
+        if (previousChoice === 'MINMAX' && !hasAnyValue) {
+            return 'MINMAX';
+        } else {
+            return 'CURVE';
+        }
+    } else {
+        console.error(
+            'Reactive capability curve choice is not valid, it should be either MINMAX or CURVE. We return MINMAX by default.'
+        );
```

Comment:
This is a tiny comment, but I dont know why we consider this case is error while we still return a normal value, i.e. 'MINMAX'